staging/src/k8s.io/client-go/tools/leaderelection/leaderelection.go

```diff
@@ -299,11 +299,26 @@ func (le *LeaderElector) release() bool {
 		RenewTime:            now,
 		AcquireTime:          now,
 	}
-	if err := le.config.Lock.Update(context.TODO(), leaderElectionRecord); err != nil {
-		klog.Errorf("Failed to release lock: %v", err)
+	todoCtx := context.TODO()
+	// update the lock itself
+	if err := le.config.Lock.Update(todoCtx, leaderElectionRecord); err == nil {
+		le.setObservedRecord(&leaderElectionRecord)
+		return true
+	} else if !errors.IsConflict(err) {
+		klog.Errorf("error updating resource lock %v: %v", le.config.Lock.Describe(), err)
+		return false
+	}
+	// Retry if conflicts.
```

Comment:
if there is a conflict, that means the live object was updated by someone else and our view of it is stale. it is incorrect/dangerous to fetch and blindly update again without re-checking if we're even still the leader

Reply:
yeah, you're right, without re-checking we may break other's leadership. I'll add a re-checking later, after the test suggestion is given. Thanks.